test/e2e/network/util.go

```diff
@@ -141,6 +141,35 @@ func execSourceIPTest(sourcePod v1.Pod, targetAddr string) (string, string) {
 	return sourcePod.Status.PodIP, host
 }
 
+// execHostnameTest executes curl to access "/hostname" endpoint on target address
+// from given Pod to check the hostname of the target destination.
+func execHostnameTest(sourcePod v1.Pod, targetAddr, targetHostname string) {
+	var (
+		err     error
+		stdout  string
+		timeout = 2 * time.Minute
+	)
+
+	framework.Logf("Waiting up to %v to get response from %s", timeout, targetAddr)
+	cmd := fmt.Sprintf(`curl -q -s --connect-timeout 30 %s/hostname`, targetAddr)
+	for start := time.Now(); time.Since(start) < timeout; time.Sleep(2 * time.Second) {
+		stdout, err = framework.RunHostCmd(sourcePod.Namespace, sourcePod.Name, cmd)
+		if err != nil {
+			framework.Logf("got err: %v, retry until timeout", err)
+			continue
+		}
+		// Need to check output because it might omit in case of error.
+		if strings.TrimSpace(stdout) == "" {
+			framework.Logf("got empty stdout, retry until timeout")
+			continue
+		}
+		break
+	}
+
+	framework.ExpectNoError(err)
+	framework.ExpectEqual(strings.TrimSpace(stdout), targetHostname)
```

Comment:
```suggestion
	pollErr := wait.PollImmediate(interval, timeout, func() (bool, error) {
		stdout, err = framework.RunHostCmd(sourcePod.Namespace, sourcePod.Name, cmd)
		if err != nil {
			framework.Logf("got err: %v, retry until timeout", err)
			return false, nil
		}
		if strings.TrimSpace(stdout) == targetHostname {
			return true, nil
		}
		framework.Logf("expected %s got %s:" targetHostname, strings.TrimSpace(stdout))
		return false, nil
		})
		framework.ExpectNoError(pollErr)
```

this is aligned with the rest of the tests, same functionality though

Reply:
I agree this code is better, but this was following same code structure as execSourceIP above, should we update both?

Reply:
Landmines everywhere :) , my experience on the old code is better not touch it as drive in pr

Reply:
I tend to agree, but I'll add a note to maybe revisit this along with other functions here

Reply:
we need to revisit, there are some tests that are not needed and a lot of duplicate funcions, break down the file per features, it is 3k lines now, ... but before code freeze is not the best moment :)